# Final Project Submission

Please fill out:
* **Student name:** Shayan Abdul Karim Khan
* **Student pace:** Self Paced
* **Scheduled project review date/time:** 
* **Instructor name:** Abhineet Kulkarni
* **Blog post URL:** 


## Problem Overview

#### Venture Opportunity:
A lot of companies are producing original video content generating huge revenues, including box office movies. In the post-pandemic world, people have filled theaters and cinema halls allowing top movies to rack up millions in box office revenues.

#### Client Insight:
Microsoft has decided to venture into the world of movie production with a new movie studio. In order to make the movie studio a success, Microsoft needs to understand what kind of movies to produce. 

#### Client Goals:
This notebook focuses on four client goals:
1. High profits 
2. Developing a fan base 
3. Developing customer loyalty 
4. Impactful market entry 

The Value proposition of how this notebook will help the client achieve these goals is listed below in the [Value Proposition](#Value-Proposition) and [Strategy](#Strategy) sections.

#### Value Proposition:
This notebook will look at multiple sources of data to understand what will allow Microsoft to generate the most promising content and set it up for long-term success. The strategy described below will be used to generate actionable insight based on two main Key Performance Indicators (KPIs):
<br>
1. *Movie Reviews*
2. *Movie Profits*


## Strategy

To answer the complex question of the secret ingredients of a successful movie, there's a myriad of aspects to consider. The different aspects analyzed will be evaluated based on the two KPIs listed above; Profits and reviews. 

Good **Movie Reviews** are important indicators of the quality of a movie nonetheless they can not solely justify an investment opportunity. Therefore **Movie Profits** will be used as an indicator of high potential investment opportunities. The two KPIs coupled together can give the client a more thorough lay of the land. 

These KPIs will be analyzed through three main factors:

1. Customer Influences
2. Competition Awareness
3. Financial Impact


The aspects that will be explored under these segments are listed below in no specific order of importance:

1. **Customer Influences:**

    It is important to understand the customer base to target and how different factors influence their behavior. This will ensure that the client produces movies that customers will be willing to invest in. Targetting the customer properly is the most important part in ensuring a thriving fan base, growing customer loyalty and having an impactful market entry. Customer saatisfaction is directly correlated with profits and better reviews which will allow long-term success.

    The factors investigated in this segment are the following:
    * Genres
    * Movie Ratings (R/PG/etc)
    * Reviews
    * Run Time
    * Initial Screening Year
    * Cast
    * Directors

2. **Competition Awareness:**

    It is critical to understand the current market competition to set the client up for success. Understanding the compeition can provide insight into barriers to entry in specific genres. It can also provide an outlook on custmer acquisation potential in scenarios where there is direct compeition with another well-established movie house.
    
    The factors investigated in this segment are the following:
    * Market share by Revenue
    * Market share by Profit
    * Market share by Genre
    * Performance in local vs foreign markets

3. **Financial Impact:**

    The financial aspect of venturing into a new business an essential area to evaluate. Understanding the investment required and the expected profits can allow the client to have realistic expectations and forecasting. 
    
    The factors investigated in this segment are the following:
    * Movie Budgets
    * Net Profits
    * Rate of Return

## Data Sources

To solve the problem, there were 5 data sources used to conduct the analysis. The data sources are listed below:

- **[Box Office Mojo](https://www.boxofficemojo.com/)**

- **[The Numbers](https://www.the-numbers.com/)**

- **[IMDB](https://www.imdb.com/)**

- **[Rotten Tomatoes](https://www.rottentomatoes.com/)**

- **[TheMovieDB](https://www.themoviedb.org/)**



These data sources were selected based on reliability and content. These are some of the most popular websites for reviews, critics and movie information.

The content and relevance of the data avaialble is dealt with in detail in the **Data Understanding** Section.

## Data Understanding

This section decyphers the data being used explores the following:

- Contents of the datasets
- Features of the datasets
- Relevance of the datasets to the project
- Relevant features of the datasets that will be used for analysis
- Relation between the different datasets
- Limitations of the datasets
- Avenues of analysis that will be pursued


### Importing Python Libraries

We will start by importing the appropriate python libraries to explore the datasets.

In [47]:
import pandas as pd #imports the pandas library as pd to work on databases
import sqlite3 as sql # imports the sqlite3 library to leverage sql with pandas
from pandasql import sqldf # imports pandas sql library
import matplotlib.pyplot as plt # importing matplotlib for visualizations 
%matplotlib inline 

We will write the lambda function for sqldf to make things easier for us throughout the notebook.

In [4]:
pysqldf = lambda q: sqldf(q, globals())

### Importing and Storing Data

#### Box Office Mojo Dataset

The **Box Office Mojo** dataset is stored in the `zippedData` folder. 

The file is called `bom.movie_gross.csv.gz`.

We can see from the extension that this dataset is a `CSV` file therefore we will use the read_csv function of pandas to explore this dataset.

In [148]:
bom_df = pd.read_csv('bom.movie_gross.csv.gz') # reading the data and converting to a dataframe

bom_df.head() # Previewing the data

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

Lets look at the overview of the data frame using the `.info()` function

In [140]:
bom_df.info() # getting the overvieww info of the dataframe records

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


There are **five** columns excluding the index column:

- `title` : This is the movie title. It is an important column because it shows which movie the record is for. This will be valuable in comparing/merging data to give more context to the analysis.
<br>

- `studio` : This gives an outlook on the competitors in the market. This will be valuable in doing competitor awareness analysis.
<br>

- `domestic_gross` : This data is particularly important because it provides the financial overview within the US market. Since there is no currency sign associated, we will assume `$` but more on that in the **Data Prepartion** section.
<br>

- `foreign_gross` : This is similar to `domestic gross` and equall valuable. It provides insights on foreign performance of a movie.
<br>

- `year` : This is the year that the movie was released. It is important to keep the year in mind so that the analysis isn't skewed. We will later evaluate which timelines to focus on for the analysis.

There are `3387` records in total in this dataset which is a good size for our analysis. 

Nonetheless, lets explore the distribution of the `year` column to understand the spread of the data to ensure that the timeline we want to look isn't skewed.

In [141]:
# getting the counts for the unique values in the year column

bom_unique_value_counts = bom_df['year'].value_counts(sort=False) 

In [142]:
print(bom_unique_value_counts)

2010    328
2011    399
2012    400
2013    350
2014    395
2015    450
2016    436
2017    321
2018    308
Name: year, dtype: int64


We can see that the data is between **2010** and **2018** with a very good distribution. This allows us to choose our timelines without fear of loosing data or skewing the results.

Also, this dataset has some values missing.

- `studio` column has 5 missing values which we can drop since it will have an insignificant on impact the total number of records but we will take a look at the records we are dropping in the **Data Preparation** section.

- `foreign gross` column has more than 1000 missing values. We will correlate this with data we find from other datasets to understand whether these movies weren't launched in foregin markets or is there a gap in our data. Also, we can use the different datasets to fill in this information if we do find an inconsistency. This is why we are using multiple dataasets. We will explore this more in the **Data Preparation** section.

**Summary:** The **Box Office Mojo** data will be very valuable in our analysis of the **Competition Awareness** and **Financial Impact** analaysis.

#### The Numbers Dataset

This dataset is stored in the `zippedData` folder. 

The file is called `tn.movie_budgets.csv.gz`.

We can see from the extension that this dataset is a `CSV` file therefore we will use the `pd.read_csv()` function of pandas to explore this dataset.

In [149]:
tn_df = pd.read_csv('tn.movie_budgets.csv.gz') # reading the data and converting to a dataframe

tn_df.head() # Previewing the data

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

Lets continue to explore further with the `.info()` function

In [145]:
tn_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


There are **six** columns in this dataset with a total of `5782` records.

Lets explore the id column to understand more about it

In [154]:
# getting the counts for the unique values in the id column

tn_df['id'].value_counts() 

1      58
63     58
61     58
60     58
59     58
       ..
86     57
85     57
84     57
83     57
100    57
Name: id, Length: 100, dtype: int64

It is difficult to guage what the `id` column is doing in this dataset. Looking at other datasets might give us better insight.

Lets check whether the `worldwide_gross` column is a sum of `domestic_gross` and foreign revenue or does it purel signify foreign revenue. For this we will compare values of the same movies between `bom_df` and `tn_df`

In [189]:
# extracting the Avengers record of tn_df
avengers_tn = tn_df.iloc[3] 

# extracting the Avengers record of bom_df by setting
# the index to title and searching for the specific title

avengers_bom = bom_df.set_index('title').loc['Avengers: Age of Ultron'] 

# printing the respective records
print('tn_df avengers record:\n', avengers_tn)
print('\nbom_df avengers record:\n', avengers_bom)

tn_df avengers record:
 id                                         4
release_date                     May 1, 2015
movie                Avengers: Age of Ultron
production_budget               $330,600,000
domestic_gross                  $459,005,868
worldwide_gross               $1,403,013,963
Name: 3, dtype: object

bom_df avengers record:
 studio                     BV
domestic_gross    459000000.0
foreign_gross       946400000
year                     2015
Name: Avengers: Age of Ultron, dtype: object


In [244]:
round(avengers_bom['domestic_gross'].astype('int32') + int(avengers_bom['foreign_gross']),-8) == round(int(avengers_tn['worldwide_gross'].replace(',','' ).replace('$','')), -8)

True

From the above True statement we can deduce that the `worldwide_gross` in `tn_df` is indeed the sum of `domestic_gross` and `foreign_gross` of the `bom_df` dataframe.

With that in mind, a brief overview of the colummns in the `tn_df` dataframe is as follows:

- `id` : This might be a secondary key of the dataset which we'll explore more when looking at the other datasets.
<br>

- `release_date` : This is the release date of the movie and wwould be valuable in performing analysis against yearly trends. This column can also be helpful to compare/group with the `bom_df` data.
<br>

- `movie` : This is the movie name which shows which movie the record is for.
<br>

- `production_budget` : This column is important in understanding the initial investments that have to be made by the client. This will also allow us to calculate the net profits and rate of return.
<br>

- `domestic-gross` : Similar to `bom_df`, this column provides insight into the revenue generated in the US market by the movie.
<br>

- `worldwide-gross` : This is slightly different than the `foreign_gross` column of `bom_df` because it adds up the domestic gross of the movie and provides the total revenue for the movie. Depending on further exploration below, we might be able to drop the `domestic_gross` and `foreign_gross` columns in `bom_df`.

An important thing to note is that `tn_df` has `5782` records while `bom_df` has `3378`. 


**Summary:** `tn_df` has similar data bins(i.e. columns) as `bom_df` but more records. The latter dataframe is not entirely composed of the same records as the former one therefore during **Data Preparation**, it will be important to join the two dataframes. 

Nonethelss, `tn_df` adds similar value to the **Financial Impact** analysis as `bom_df`, increasing the sample size for our analaysis.

## Data Preparation

## Data Analysis

## Results

## Conclusion

## Recommendation

In [ ]:
# Your code here - remember to use markdown cells for comments as well!